app/models/merge_request.rb

```diff
@@ -31,4 +31,29 @@ def mentioned_issue_ids
       (value || '').scan(ISSUE_ID_REGEXP)
     end.uniq
   end
+
+  def fixed_issue_ids(fixing_pattern)
+    fixed_issue_regexp = fixing_pattern + '\s+' + ISSUE_ID_REGEXP.source
+    [description, title].flat_map do |value|
+      (value || '').scan(/#{fixed_issue_regexp}/i)
+    end.uniq
+  end
+
+  def update_mentioned_issues_status
+    redmine_user_id = ENV['REDMINE_MERGE_REQUEST_LINKS_GITLAB_REDMINE_USER_ID']
+    after_merge_status = ENV['REDMINE_MERGE_REQUEST_LINKS_AFTER_MERGE_STATUS']
+    fixing_pattern = ENV['REDMINE_MERGE_REQUEST_LINKS_FIXING_KEYWORD_PATTERN']
+    if state != 'merged' || redmine_user_id.blank? || after_merge_status.blank?
+      return
+    end
+    issue_ids = fixing_pattern.present? ? fixed_issue_ids(fixing_pattern) : mentioned_issue_ids
+    issue_ids.map do |match|
+      issue = Issue.find_by_id(match[0])
+      if issue.present?
+        issue.init_journal(User.find(redmine_user_id))
+        issue.status = IssueStatus.find_by_name(after_merge_status)
+        issue.save!
```

Comment:
What happens when the state transition is not allowed (i.e. has been deselected in the tracker's workflow settings)?

Reply:
If the status is not allowed in workflow, the status is changed anyway. Is it sufficient for now or should we add some check?

Reply:
I was just looking at [the way Redmine handles status updates via commit messages](https://github.com/redmine/redmine/blob/c04ab14ed4843baa0a7d55fe61a7377bb4909880/app/models/changeset.rb#L246). Maybe we can follow the same approach and log validation error instead of raising an exception when the issue cannot be saved. That way the merge request gets updated correctly even if an issue update fails.